File: backend/src/test/java/br/ufpr/sept/so2/shared/domain/valueobject/EmailTest.java
package br.ufpr.sept.so2.shared.domain.valueobject;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class EmailTest {

    @Test
    void identificaEmailInstitucional() {
        assertTrue(Email.of("aluno@ufpr.br").isInstitutional());
        assertFalse(Email.of("aluno@gmail.com").isInstitutional());
    }
}

File: backend/src/test/kotlin/br/ufpr/sept/so2/shared/domain/valueobject/EmailTest.kt
package br.ufpr.sept.so2.shared.domain.valueobject

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe

class EmailTest : StringSpec({
    "identifica email institucional" {
        Email.of("aluno@ufpr.br").isInstitutional() shouldBe true
        Email.of("aluno@gmail.com").isInstitutional() shouldBe false
    }
})
